### ISSUE TITLE
Coloring of CSV documents not optimal

### ISSUE THREAD

**Issue by dinkumoil** [BdR76/CSVLint#8]
## Description

This issue has 2 aspects. If you decide to solve number 1. then number 2. is automatically obsolet.

1. When placing the caret to a line of a CSV document, all color markings of CSV columns disappear. The background color is changed to Npp's color for highlighting the current line (where the caret is placed). This reduces the usefulness of the plugin. It would be more helpful to use the **foreground** color (i.e. the color of the characters) for color marking of CSV columns. This way the line where the caret is placed would still contain color markings.
 
2. The default color set for coloring column content of CSV documents contains for the 1st, 9th, ... column a color which is nearly identical to Npp's color for highlighting the current line (where the caret is placed). Thus it is nearly impossible to distinguish the current line from that said columns.

## Applies to

Plugin version 0.4.1

**Comment by dinkumoil** [comment]
Thank you for providing a detailed documentation for your plugin! 

I downloaded the most recent version (v0.4.2) and tested it with portable installations of Notepad++ v8.1.5 (both 32 bit and 64 bit versions). When Notepad++ starts up for the first time after plugin's installation, it creates a _CSVLint.xml_ file in directory `plugins\Config` with the following content:

```
<?xml version="1.0" encoding="utf-8"?>
<NotepadPlus>
	<Languages>
		<Language name="CSVLint" ext="csv" commentLine="#" commentStart="#[" commentEnd="]#">
			<Keywords name="instre1" />
			<Keywords name="instre2" />
			<Keywords name="type1" />
			<Keywords name="type2" />
			<Keywords name="type3" />
			<Keywords name="type4" />
			<Keywords name="type5" />
			<Keywords name="type6" />
		</Language>
	</Languages>
	<LexerStyles>
		<LexerType name="CSVLint" desc="CSV Linter and validator" excluded="no" ext="">
			<WordsStyle styleID="0" name="Default" fgColor="000000" bgColor="FFFFFF" fontName="" fontStyle="0" />
			<WordsStyle styleID="1" name="ColumnColor1" fgColor="000000" bgColor="E0E0FF" fontName="" fontStyle="0" />
			<WordsStyle styleID="2" name="ColumnColor2" fgColor="000000" bgColor="FFFF80" fontName="" fontStyle="0" />
			<WordsStyle styleID="3" name="ColumnColor3" fgColor="000000" bgColor="FFE0FF" fontName="" fontStyle="0" />
			<WordsStyle styleID="4" name="ColumnColor4" fgColor="000000" bgColor="80FF80" fontName="" fontStyle="0" />
			<WordsStyle styleID="5" name="ColumnColor5" fgColor="000000" bgColor="FFB0FF" fontName="" fontStyle="0" />
			<WordsStyle styleID="6" name="ColumnColor6" fgColor="000000" bgColor="FFC0C0" fontName="" fontStyle="0" />
			<WordsStyle styleID="7" name="ColumnColor7" fgColor="000000" bgColor="32FFBE" fontName="" fontStyle="0" />
			<WordsStyle styleID="8" name="ColumnColor8" fgColor="000000" bgColor="FFD040" fontName="" fontStyle="0" />
		</LexerType>
	</LexerStyles>
</NotepadPlus>
```

Unfortunately now there is only **one** color scheme in the file.

**Comment by BdR76** [comment]
CSV Lint was [updated to v0.4.3](https://github.com/BdR76/CSVLint/releases), it's automatically available in `Notepad++ v8.1.9.1` or newer, under the `Plugins Admin..` menu item.

* Default CSVLint.xml now has [4 color presets](https://github.com/BdR76/CSVLint/tree/master/docs#Syntax-highlighting-colors), 2 for normal 2 for dark mode
* At first time start-up, the plugin tries to detect if running in Dark Mode and sets color preset accordingly
* Settings dialog now has `Colors` button to select any of the 4 color presets (requires Notepad++ a restart)

**Comment by dinkumoil** [comment]
Thank you very much for this update, works perfectly!
